test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -0,0 +1,202 @@
+        jmxc = null;
+
+        ssProxy = StorageService.instance;
+        msProxy = MessagingService.instance();
+        streamProxy = StreamManager.instance;
+        compactionProxy = CompactionManager.instance;
+        fdProxy = (FailureDetectorMBean) FailureDetector.instance;
+        cacheService = CacheService.instance;
+        spProxy = StorageProxy.instance;
+        hhProxy = HintedHandOffManager.instance;
+        gcProxy = new GCInspector();
+        gossProxy = Gossiper.instance;
+        memProxy = ManagementFactory.getMemoryMXBean();
+        runtimeProxy = ManagementFactory.getRuntimeMXBean();
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        // nothing to close. no-op
+    }
+
+    @Override
+    // overrides all the methods referenced mbeanServerConn/jmxc in super
+    public EndpointSnitchInfoMBean getEndpointSnitchInfoProxy()
+    {
+        return new EndpointSnitchInfo();
+    }
+
+    @Override
+    public CacheServiceMBean getCacheServiceMBean()
+    {
+        return cacheService;
+    }
+
+    @Override
+    public void repairAsync(final PrintStream out, final String keyspace, Map<String, String> options) throws IOException
+    {
+        RepairRunner runner = new RepairRunner(out, ssProxy, keyspace, options);
+        try
+        {
+            ssProxy.addNotificationListener(runner, null, null);
+            runner.run();
+        }
+        catch (Exception e)
+        {
+            throw new IOException(e) ;
+        }
+        finally
+        {
+            try
+            {
+                ssProxy.removeNotificationListener(runner);
+            }
+            catch (Throwable e)
+            {
+                out.println("Exception occurred during clean-up. " + e);
+            }
+        }
+    }
+
+    @Override
+    // simply resume bootstrap, instead of streaming the JMX notification to stdout
+    public void resumeBootstrap(PrintStream out) throws IOException
+    {
+        if (ssProxy.resumeBootstrap())
+        {
+            out.println("Resuming bootstrap");
+        }
+        else
+        {
+            out.println("Node is already bootstrapped.");
+        }
+    }
+
+    // dtest runs with disable_mbean_registration. No ColumnFamilyStoreMBean is registered, thus not supporting get the MBean.
+    @Override
+    public ColumnFamilyStoreMBean getCfsProxy(String ks, String cf)
+    {
+        return Keyspace.open(ks).getColumnFamilyStore(cf);
+    }
+
+    @Override
+    // The below methods are only used by the commands (i.e. Info, TableHistogram, TableStats, etc.) that display informations. Not useful for dtest, so disable it.
+    public Object getCacheMetric(String cacheType, String metricName)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, ColumnFamilyStoreMBean>> getColumnFamilyStoreMBeanProxies()
+    {
+        throw new UnsupportedOperationException();
```

Comment:
FYI

```
return StreamSupport.stream(Keyspace.all().spliterator(), false)
                     .flatMap(ks -> ks.getColumnFamilyStores().stream()
                                      .<Map.Entry<String, ColumnFamilyStoreMBean>>map(store -> ImmutablePair.of(ks.getName(), store)))
                     .iterator();
```

Reply:
Since it is only used for output stats, I will keep it as unsupported. But good to know how to make a columnFamilyStoreIterator.